src/components/Api.js

```diff
@@ -18,19 +18,13 @@ const fetchAndSortTodos = async (table, sortOptions) => {
   const todos = data.records.map((todo) => ({
     title: todo.fields.title,
     id: todo.id,
-    createdTime: todo.createdTime,
-    isChecked: todo.done,
+    createdTime: new Date(todo.createdTime),
+    isChecked: todo.fields.done ?? false,
   }));
 
-  const sortedTodos = todos.sort((a, b) => {
-    if (a.createdTime > b.createdTime) {
-      return 1;
-    }
-    if (a.createdTime < b.createdTime) {
-      return -1;
-    }
-    return 0;
-  });
+  const sortedTodos = sortOptions !== undefined
+  ? todos.sort((a, b) => a.createdTime - b.createdTime)
+  : todos;
 
   return sortedTodos;
```

Comment:
because the fields have already been pre-sorted by the api call (and its query parameters), you don't need this frontend sort to happen at all on initial fetch. You only need the frontend sort in response to the button click
```suggestion
  return todos;
```

Reply:
Very good thing to keep in mind. Thank you for this feedback and changed per your suggestions!